brew.py

```diff
@@ -1,104 +1,126 @@
+                check_installed_format = "arch --x86_64 " + check_installed_format
+            isInstalled = subprocess.call(
+                check_installed_format % (pkg),
+                shell=True, stdin=devnull, stdout=devnull, stderr=devnull, cwd=cwd)
+            if isInstalled != 0:
+                self._log.info("Installing %s" % pkg)
+                result = self._invokeShellCommand(
+                    install_format % (pkg))
                 if result != 0:
-                    log.warning('Failed to install file [%s]' % f)
+                    self._log.warning('Failed to install [%s]' % pkg)
                     return False
+            else:
+                self._log.info("%s already installed" % pkg)
             return True
 
-    def _bootstrap(self, cmd):
-        with open(os.devnull, 'w') as devnull:
-            stdin = stdout = stderr = devnull
-            subprocess.call(cmd, shell=True, stdin=stdin, stdout=stdout, stderr=stderr,
-                            cwd=self._context.base_directory())
+    def _brewfile(self, brew_files):
+        # this directive has opposite defaults re stdin/stderr/stdout
+        self._defaults['stdin'] = self._defaults.get('stdin', True)
+        self._defaults['stderr'] = self._defaults.get('stderr', True)
+        self._defaults['stdout'] = self._defaults.get('stdout', True)
+        self._bootstrap_brew()
+        self._bootstrap_cask()
+        for f in brew_files:
+            self._log.info("Installing from file %s" % f)
+            cmd = "brew bundle --verbose --file=%s" % f
+            result = self._invokeShellCommand(cmd)
 
-    def _bootstrap_brew(self):
-        link = "https://raw.githubusercontent.com/Homebrew/install/master/install"
-        if platform.system() == 'Linux':
-            link = "https://raw.githubusercontent.com/Linuxbrew/install/master/install"
+            if result != 0:
+                self._log.warning('Failed to install file [%s]' % f)
+                return False
+        return True
 
-        cmd = """hash brew || ruby -e "$(curl -fsSL {0})";
-              brew update""".format(link)
-        self._bootstrap(cmd)
+    def _bootstrap_brew(self):
+        self._log.info("Installing brew")
+        link = "https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh"
+        cmd = """[[ $(command -v brew) != "" ]] || /bin/bash -c "$(curl -fsSL {0})" """.format(
+            link)
+        if self._defaults.get(self._forceIntelOption, False) == True:
+            cmd = "arch --x86_64 " + cmd
+        return subprocess.call(cmd, shell=True, cwd=self._context.base_directory()) == 0
 
     def _bootstrap_cask(self):
-        self._bootstrap_brew()
-        cmd = "brew tap caskroom/cask"
-        self._bootstrap(cmd)
+        self._log.info("Installing cask")
+        cmd = "brew tap homebrew/cask"
+        if self._defaults.get(self._forceIntelOption, False) == True:
+            cmd = "arch --x86_64 " + cmd
```

Comment:
I am not sure how this works. The final command would be `arch --x86_64 brew tap homebrew/cask`. Could you please point me where I can read about the issue with ARM? I tried grepping `man brew`, but didn't find it.